### PR TITLE
"assign to me" feature

### PR DESCRIPTION
**Pull request by philippe-levan**
Hi,

I suggest to add an "assign to me" feature on the contextual menu of each card.
Here is a PR.

Warnings : 

* I'm completely noob in angular. Is is possible to read this PR carefully ?
* I just add a translation for french. I don't know how you manage translations.

Best regards,
Philippe

PS : here's a screnshot.
image

### CODE REVIEW

File: js/controller/BoardController.js [hunk 1]
```diff
@@ -200,6 +200,27 @@ app.controller('BoardController', function ($rootScope, $scope, $stateParams, St
 		CardService.archive(card);
 		StackService.removeCard(card);
 	};
+    $scope.isCurrentUserAssigned = function (card) {
```
Comment (juliushaertl) [inline js/controller/BoardController.js]: Please use tabs instead of spaces.

File: l10n/fr.js [hunk 2]
```diff
@@ -96,6 +96,8 @@ OC.L10N.register(
     "Shared boards" : "Tableaux partagés",
     "View more" : "Voir plus",
     "Move board to archive" : "Déplacer le tableau vers l'archive",
-    "Create a new board" : "Créer un nouveau tableau"
+    "Create a new board" : "Créer un nouveau tableau",
```
Comment (juliushaertl) [inline l10n/fr.js]: Please remove the changes to the translation files, those are added by transifex.

File: js/controller/BoardController.js [hunk 3]
```diff
@@ -200,6 +200,27 @@ app.controller('BoardController', function ($rootScope, $scope, $stateParams, St
...
+            return obj.participant.uid === OC.getCurrentUser().uid;
+        });
+		return userList.length === 1;
+    };
+    $scope.cardAssignToMe = function (card) {
+        CardService.assignUser(card, OC.getCurrentUser().uid)
+            .then(
+                function() {StackService.updateCard(card);}
+            );
+        $('.popovermenu').addClass('hidden');
```
Comment (juliushaertl) [inline js/controller/BoardController.js]: We should avoid using jquery in that way. Maybe you can have a look why the popovermenu is not closed in the directive that is opening it (see js/directive/appPopoverMenuUtils.js)

File: l10n/fr.js (hunk 2 above)
Comment (philippe-levan) [reply l10n/fr.js]: fixed in 02926d93cb1864e16c4c6d628af4afdadb17bf5b

File: js/controller/BoardController.js (hunk 1 above)
Comment (philippe-levan) [reply js/controller/BoardController.js]: fixed by 02c1807f622e81212a97843779a90ed9641e9a7c

File: js/controller/BoardController.js (hunk 3 above)
Comment (philippe-levan) [reply js/controller/BoardController.js]: Hi,

I don't manage to make it work with appPopoverMenuUtils.js.
I believe the click event is stopped by several stopPropagation at different levels.

I don't manage to make it work without entering deeply in the application code. I believe I don't have the necessary JS skills to make it work without breaking all the app :-)

Can you help me debug that point ?

Best regards,
Philippe

**Comment by philippe-levan** [comment]
I squashed all the commits into a single one with my signed-off-by

**Comment by philippe-levan** [comment]
Hi @juliushaertl ,

I made the modifications I'm able to do right now. I'm blocked with your review on JQuery (see https://github.com/nextcloud/deck/pull/544#discussion_r204536453).

Best regards,
Philippe

**Comment by juliushaertl** [comment]
> I made the modifications I'm able to do right now. I'm blocked with your review on JQuery (see #544 (comment)).

Ok, I'd be fine to keep it like this for now, I can have a look into that afterwards. Mind to just add a TODO comment there?

**Comment by philippe-levan** [comment]
I just added the TODO comments

**Comment by juliushaertl** [comment]
Failing CI is fine, since it is about the missing signoff message on the merge commit and on a comment only commit.

**Comment by juliushaertl** [comment]
Thank you very much for your great contribution @philippe-levan If you are interested in getting involved further, be sure to checkout the list of [good first issues](https://github.com/nextcloud/deck/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22).

**Pull request merged**